server/src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -84,5 +94,65 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.storeMessage(message);
 
     response.sendRedirect("/userpage?user=" + user);
+    String user = userService.getCurrentUser().getEmail();
+    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    List<String> imageBlobUrls = getUploadedFileUrl(request, "image");
+    if (imageBlobUrls != null) {
+      for (String u : imageBlobUrls) {
+        text += String.format("<br/><img src=\"%s\"/>", u);
+      }
+    }
+
+    Message message = new Message(user, text);
+    datastore.storeMessage(message);
+
+    response.sendRedirect("/user-page.html?user=" + user);
+  }
+
+  private List<String> getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL.
+    // (devserver)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      System.out.printf(
+          "Assuming this is the devserver and nothing was uploaded; blobKeys: %s%n", blobKeys);
+      return null;
+    }
+
+    for (BlobKey blobKey : blobKeys) {
+      BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+      if (blobInfo.getSize() == 0) {
+        System.out.printf("deleting blobkey [%s] because it's empty", blobKey);
+        blobstoreService.delete(blobKey);
+      }
+    }
+
+    // Use ImagesService to get a URL that points to the uploaded file.
+    ImagesService imagesService = ImagesServiceFactory.getImagesService();
+    List<String> imageBlobUrls = new ArrayList<>();
+    for (BlobKey blobK : blobKeys) {
+      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobK);
+      try {
+        // getting the image URL to the uploaded file
```

Comment:
nit: Capitalize the first letter of your comments.

`// Getting the image URL to the uploaded file`